ateliermecanique-ws/src/main/java/com/champlain/ateliermecaniquews/customerinvoicemanagementsubdomain/businesslayer/CustomerInvoiceServiceImpl.java

```diff
@@ -53,4 +55,26 @@ public List<CustomerInvoiceResponseModel> getAllInvoicesByCustomerId(String cust
         }
         return invoiceResponseModels;
     }
+
+    @Override
+    public CustomerInvoiceResponseModel addInvoiceToCustomerAccount(String customerId, CustomerInvoiceRequestModel customerInvoiceRequestModel) {
+        User customerAccount = userRepository.findUserByUserIdentifier_UserId(customerId);
+
+        if(customerAccount == null) {
+            log.warn("Customer account not found for customer ID: {}", customerId);
+            return null;
+        }
+
+        CustomerInvoice invoice = new CustomerInvoice();

```

Comment:
If you have created a RequestMapper, this is were you would use it, it would take one line instead of 6